core-api/src/main/java/com/optimizely/ab/config/audience/Condition.java

```diff
@@ -17,15 +17,25 @@
 package com.optimizely.ab.config.audience;
 
 import com.optimizely.ab.config.ProjectConfig;
+import com.optimizely.ab.optimizelydecision.DecisionReasons;
+import com.optimizely.ab.optimizelydecision.OptimizelyDecideOption;
 
 import javax.annotation.Nullable;
+import java.util.List;
 import java.util.Map;
 
 /**
  * Interface implemented by all conditions condition objects to aid in condition evaluation.
  */
 public interface Condition<T> {
 
+    @Nullable
+    Boolean evaluate(ProjectConfig config,
+                     Map<String, ?> attributes,
+                     List<OptimizelyDecideOption> options,
+                     DecisionReasons reasons);
+
     @Nullable
     Boolean evaluate(ProjectConfig config, Map<String, ?> attributes);
```

Comment:
Do we need to keep this signature? Feels like we only need it for testing. Since "we" own the calling of this method we can pass defaults there.